File: PRACTICAS/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/especulador.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "jugador"
module ModeloQytetet

	class Especulador < Jugador
  
		def initialize jugador, f
			super(jugador.nombre, jugador.encarcelado, jugador.saldo, jugador.cartaLibertad, jugador.casillaActual, jugador.propiedades)
			@fianza = f
		end
  
		def get_factor_especulador
			return 2
		end
	
		def convertirme fianza
			self
		end
	
		def pagar_impuestos cantidad
			puts "PAGAS #{cantidad/2}************************************************************"
			modificar_saldo(-cantidad/2);
		end
	
		def pagar_fianza f
			if(tengo_saldo(f))
				modificar_saldo(-f)
				return true
			else
				return false
			end
		end
	
		def ir_a_carcel casilla
			begin
				puts "Quieres pagar una fianza para evitar ir a la carcel? <0, SI> <1, NO>\n"
				eleccion = gets.chomp
				numero = eleccion.to_i
			end while(numero!=0 && numero!=1)
		
			if (numero == 0 && pagar_fianza(@fianza))
				puts "Has pagado para no entrar en la carcel. Seguirás en la misma casilla\n"
			else
				@casillaActual = casilla
				@encarcelado = true
			end
		end
	
		def to_s
			s = "\n      Jugador tipo ESPECULADOR.\n      factorEspeculador: #{get_factor_especulador()}\n      Fianza: #{@fianza} \n      nombre: #{@nombre} \n      saldo: #{@saldo} \n      encarcelado: #{@encarcelado} \n      cartaLibertad: #{@cartaLibertad} \n      casillaActual: #{@casillaActual}      propiedades: \n "
			for prop in @propiedades
				s += prop.to_s
				s += "\n"
			end
			s
		end
    
	end

end

File: PRACTICAS/QytetetJava/src/modeloqytetet/Especulador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.Scanner;

/**
 *
 * @author jotajota98
 */
public class Especulador extends Jugador {
    static int FactorEspeculador = 2;
    private int fianza;
    
    protected Especulador(Jugador jug, int f)
    {
        super(jug);
        fianza = f;
    }
    
    public int getFianza()
    {
        return fianza;
    }
    
    private boolean pagarFianza(int cantidad)
    {
        if ( tengoSaldo(cantidad) )
        {
            modificarSaldo(-cantidad);
            return true;
        }
        else
            return false;
    }
    
    @Override
    protected void pagarImpuestos(int cantidad)
    {
        modificarSaldo(-cantidad/2);
    }
    
    @Override
    protected void irACarcel(Casilla casilla)
    {
        Scanner scan = new Scanner(System.in);
        int opcion;
        
        do{
            System.out.println("Quieres pagar una fianza para evitar ir a la carcel? <0, SI> <1, NO>\n");
            opcion = scan.nextInt();
        }while (opcion != 0 && opcion != 1);
        
        if (opcion == 0 && pagarFianza(fianza))
        {
            System.out.println("Has pagado para no entrar en la carcel. Seguirás en la misma casilla\n");
        }
        else
        {
            setCasillaActual(casilla);
            setEncarcelado(true);
        }
    }
    
    @Override
    protected Especulador convertirme(int fianza)
    {
        return this;
    }
    
    @Override
    public String toString() {
        String resultado = "\n      Jugador tipo ESPECULADOR.\n      factorEspeculador:" + FactorEspeculador + "\n      Fianza: " + fianza + "\n";
        resultado += super.toString();
        return resultado;
    }
}
